### ISSUE TITLE
new dash quality setting doesnt work

### ISSUE THREAD

**Issue by rezad1393** [iv-org/invidious#1630]
I am testing with this instance 
https://invidious.snopyta.org

I have set the format to dash and quality to 720p.
but I get the 360p version of video. which you can also test to see the resolution of video played by right clicking and the video and getting an snapshot which would be in 360p.
this is a example link
https://invidious.snopyta.org/watch?v=OuF9weSkS68

**Comment by unixfox** [comment]
I can't replicate the bug on my instance: https://yewtu.be/watch?v=OuF9weSkS68.

You may have a lower video quality if the instance is overloaded and can't keep up with the amount of visitors trying to load a dash video at the same time. In this case the instance will serve you a lower video quality.

In the end the dash setting is just a preferred video quality not the one that you will have in every case, Invidious will just try its best to serve you the preferred video quality but if it can't then it will serve you the quality that it can process fast enough.

**Comment by rezad1393** [comment]
> I can't replicate the bug on my instance: https://yewtu.be/watch?v=OuF9weSkS68.
> 
> You may have a lower video quality if the instance is overloaded and can't keep up with the amount of visitors trying to load a dash video at the same time. In this case the instance will serve you a lower video quality.
> 
> In the end the dash setting is just a preferred video quality not the one that you will have in every case, Invidious will just try its best to serve you the preferred video quality but if it can't then it will serve you the quality that it can process fast enough.

I forgot to add that I use the "proxy video option"
and I CAN reproduce it on you instance.

**Comment by unixfox** [comment]
The proxy video option doesn't matter with dash because it is already being proxied.

Please create a small video in order to prove your facts.

**Comment by rezad1393** [comment]
I think I see what you mean now.
it seem that dash is actually variable rate again (like youtube)
even though I set the dash quality to 720p still it start with 360 and on your instance goes to 720p after some seconds.
but on the other instance (maybe as you say) it doesnt have the power so it lowers it to 360p which would seem random to me.
which I saw but I was not sure that was 720p or not (firefox doesnt provide a onscreen tool to see video info, or does it? I find the screenshot way by chance)
so far I am with you and understand your answer.

but I thought this 
https://github.com/iv-org/invidious/pull/1572
meant that the dash quality was not gonna change mid stream?
it there a way to do that?
as in: set a max dash quality in setting (like 1080p) and try to see if the google has that resolution and then play it without lowering the quality from that resolution even with bad internet and server load?
I know dash is designed for the opposite but the simple video load in firefox is horrible (as I said here)
https://github.com/iv-org/invidious/issues/1576

**Comment by FireMasterK** [comment]
#1619 aims to fix this, but needs changes
yewtu.be doesn't have this issue as it's running an older version of Invidious

**Comment by unixfox** [comment]
> yewtu.be doesn't have this issue as it's running an older version of Invidious

That's a false statement, yewtu.be is running the latest commit (https://github.com/iv-org/invidious/commit/28dac81a90c9c345f926369da94854c76b212d36)
From https://yewtu.be/api/v1/stats:
```json
{
  "version": "2.0",
  "software": {
    "name": "invidious",
    "version": "-28dac81",
    "branch": "master"
  }
}
```

**Comment by FireMasterK** [comment]
> That's a false statement, yewtu.be is running the latest commit ([28dac81](https://github.com/iv-org/invidious/commit/28dac81a90c9c345f926369da94854c76b212d36))
> From https://yewtu.be/api/v1/stats:
> 
> ```json
> {
>   "version": "2.0",
>   "software": {
>     "name": "invidious",
>     "version": "-28dac81",
>     "branch": "master"
>   }
> }
> ```

My bad, I assumed it was out of date due to it missing the date in the prefix of the version. (which was broken for docker until #1538) I should have checked the commit hash instead.

However, I did find something:

Invidious seems to be broken on chromium, due to a videojs muxing error? I tried using Brave Nighly and Google Chrome for this.

![image](https://user-images.githubusercontent.com/20838718/103487729-83ffb400-4e2d-11eb-8799-341e0287a906.png)

There are no network errors (except for a cancelled request, which is normal), so it's not caused due to an overloaded instance, and it seems to affect all instances which I've tested so far.

I could not replicate this on Firefox and Bromite though.
